src/plugins/workspace/server/plugin.ts

```diff
@@ -15,12 +15,29 @@ import { WorkspaceClient } from './workspace_client';
 import { registerRoutes } from './routes';
 import { WORKSPACE_CONFLICT_CONTROL_SAVED_OBJECTS_CLIENT_WRAPPER_ID } from '../common/constants';
 import { WorkspaceConflictSavedObjectsClientWrapper } from './saved_objects/saved_objects_wrapper_for_check_workspace_conflict';
+import { cleanWorkspaceId, getWorkspaceIdFromUrl } from '../../../core/server/utils';
 
 export class WorkspacePlugin implements Plugin<{}, {}> {
   private readonly logger: Logger;
   private client?: IWorkspaceClientImpl;
   private workspaceConflictControl?: WorkspaceConflictSavedObjectsClientWrapper;
 
+  private proxyWorkspaceTrafficToRealHandler(setupDeps: CoreSetup) {
```

Comment:
Wouldnt this be better as a part of core and not a plugin?

Reply:
We need to put this code here because we want it to be controlled by our feature flag `workspace.enabled`. So only when workspace is enabled will the function take effect.

Reply:
Doesnt core have access to config values?

Reply:
Yes, core has access to config values but it is a plugin's behavior to proxy the traffic or not. Core has provided the capability to let plugins to `registerOnPreRouting` and core should not depend on a plugin's config to enable/disable any functionality.